test/fcntl/test_fcntl_dup.c

```diff
@@ -0,0 +1,29 @@
+#include <unistd.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <errno.h>
+#include <assert.h>
+
+int
+main() {
+    int fd, dupfd, status, bytes;
+    fclose(stdin);
+    fd = open("a.txt", O_CREAT | O_WRONLY, 0644);
+    assert(fd == 0);
+    printf("fd: %d errno: %d\n", fd, errno);
+    bytes = write(fd, "hi there!\n", 11);
+    assert(bytes == 11);
+    printf("fd: %d errno: %d bytes: %d\n", fd, errno, bytes);
+
+    dupfd = fcntl(fd, F_DUPFD, 0);
+    assert(dupfd != fd);
+    printf("dupfd: %d\n", dupfd);
+    status = close(dupfd);
+    printf("status: %d\n", status);
+    assert(status == 0);
+    bytes = write(fd, "hi again!", 10);
+    printf("fd: %d errno: %d bytes: %d\n", fd, errno, bytes);
+    assert(bytes == 10);
+    status = close(fd);
+    assert(status == 0);
```

Comment:
It looks like we already have some testing of `F_DUPFD` in `test/fcntl/test_fcntl.c`.  Should we try to extent that rather than adding a new test..  Or if now should we delete the `F_DUPFD` testing from test/fcntl/test_fcntl.c now that we have this new test?

Reply:
I think extending the testing there is reasonable.